### PR TITLE
Add Python3 Compatibility

### PR DESCRIPTION
**Pull request by terricain**
Hey, I thought seeing as Twisted has ported a large majority of code to Python3 this library could benefit from it as well.

First of all, sorry for the large patch of small changes. I've converted the main library to be compatible with both Python2 and Python3, hope that's OK :wink:. I haven't updated the `examples`, `setup.py` to include the six dependancy (though Twisted also depends on it) or the `.travis.yml` to run both Python2 and Python3 tests, if your happy with this PR I'll also commit those changes.

Have run the test cases for both Python2 and Python3, all unit tests that use RabbitMQ pass (the other ones were skipped.

## Test Results: 
[txamqp-test-python2.txt](https://github.com/txamqp/txamqp/files/1123133/txamqp-test-python2.txt)
[txamqp-test-python3.txt](https://github.com/txamqp/txamqp/files/1123135/txamqp-test-python3.txt)

As far as I'm aware the code has not changed functionally (excluding the 2 method that handle actual binary data), from the end users view it should act exactly as expected.

### codec.py
* Swapped out StringIO with BytesIO
* Created dedicated enc/dec functions to handle actual binary data, then the current enc/dec string methods take care of strings
* Changed some strings in enc/dec table methods to byte strings for byte == byte comparisons
* Swapped out xrange for range (could of used six.moves.range)

### connection.py
* Swapped out StringIO with BytesIO
* Changed relative import syntax
* Used six's cross version metaclass compatibility
* Changed the enc/dec methods in Method to use encode_longbytes and decode_longbytes

### delegate.py
* Changed relative import syntax

### endpoint.py
* URI.fromBytes really wants bytes not unicode so did some .encodes and .decodes to keep it happy

### protocol.py
* Swapped out StringIO with BytesIO
* Changed except syntax to except as
* Swapped out xrange for range (could of used six.moves.range)
* Changed FrameReceiver.__buffer to a bytestring so then += still works
* Changed "AMQP" to a bytestring b"AMQP"
* In readContent I replaced StringIO with six.StringIO as io.StringIO is not the same as StringIO.StringIO in Python2 and this retains expected behaviour

### spec.py
* Replaced module creation code with the types.ModuleType way which is cross version
* Replaced "dict.has_key" with "key in dict"
* Changed exec to its functional equivelent
* Changed print to its functional equivelent

### testing.py
* Changed iteritems to items
* Changed print to its functional equivelent
* Changed except syntax to except as

### testlib.py
* Surrounded filter with list as the result gets indexed and Python3's filter returns an iterator

### test/test_queue.py
* Changed print to its functional equivelent, arguably the print in this file is unnecessary

### test/test_testlib.py
* Changed print to its functional equivelent, arguably the print in this file is unnecessary

### CODE REVIEW

**Comment by esteve** [comment]
@terrycain wow, thank you so much, the changes look great. @freeekanayaka any thoughts? I'll merge the PR at the end of the day and if anything comes up, we can just fix it later.

Thanks again @terrycain !

**Comment by terricain** [comment]
I updated the simple example, fixing "print"s and I replaced the sys.argv stuff with argparse as I thought that was nicer and does all the arg logic for you.

**Comment by terricain** [comment]
Updated thrift examples to be compatible with Python3 syntax. 

They currently wont work with Python3 until Thrift 0.11.0 is released. Shouldn't be too long as according to their JIRA, there's only 22 issues left in that release. Specifically we're waiting on at least [THRIFT-4009](https://issues.apache.org/jira/browse/THRIFT-4009)

File: .travis.yml [hunk 1]
```diff
@@ -2,7 +2,12 @@ sudo: false
 dist: trusty
 language: python
 python:
-  - 2.7
-services: rabbitmq
-install: pip install -r requirements.txt
-script: cd src; TXAMQP_BROKER=RABBITMQ trial txamqp
+  - "2.7"
+  - "3.3"
+  - "3.4"
```
Comment (freeekanayaka) [inline .travis.yml]: FWIW I don't think we need 3.3 and 3.4. I believe they are becoming obsolete these days and have very few users. Other projects are dropping support for 3.3 and 3.4 too.

**Review by freeekanayaka — APPROVED** [review]
Yeah thanks! @esteve I skimmed the changes and they look fine, I'd personally just drop 3.3 and 3.4 from the test matrix. I could have closer look the coming weeks and if everything is fine, cut a new release.

File: .travis.yml (hunk 1 above)
Comment (terricain) [reply .travis.yml]: Sure and I definitely agree, will commit the change later.

**Comment by terricain** [comment]
If this is all good and gets merged. Do you have any issue with me pep8'ing most of the codebase so the indents are consistent, no trailing semicolons etc...?

**Comment by freeekanayaka** [comment]
Re pep8 & co, no problem for me, thanks.

**Comment by esteve** [comment]
@terrycain yeah, go for it, thanks!

**Review by esteve — APPROVED** [review]

**Comment by takluyver** [comment]
Working on porting Graphite, a downstream project which uses txamqp. Don't forget to do a new release! Thanks :-)

**Comment by esteve** [comment]
@takluyver done https://pypi.python.org/pypi/txAMQP/0.8.0 :-) Sorry it took so long!

**Comment by takluyver** [comment]
No problem, thanks for the release. :-)

**Pull request merged**